### PR TITLE
Added cave spider to MobSpawner

### PR DESCRIPTION
**Pull request by Seadragon91**
Did no one missed the small, dangerous, poisonous spiders? :wink:
Had missed them in my [mazes](https://github.com/Seadragon91/MazeGenerator) :smile:

### CODE REVIEW

**Comment by Seadragon91** [comment]
Used the code from the spider, just changed the x and z from 2 to 1.

File: src/MobSpawner.cpp [hunk 1]
```diff
@@ -236,6 +236,32 @@ bool cMobSpawner::CanSpawnHere(cChunk * a_Chunk, int a_RelX, int a_RelY, int a_R
...
 			}
+
+			case mtCaveSpider:
+			{
+				bool CanSpawn = true;
+				bool HasFloor = false;
+				for (int x = 0; x < 1; ++x)
+				{
+					for (int z = 0; z < 1; ++z)
+					{
```
Comment (madmaxoft) [inline src/MobSpawner.cpp]: These two loops are not needed at all, they basically just say 'int x = 0;` and `int z = 0`, since they are never repeated.

**Comment by madmaxoft** [comment]
It would have been better if you copied the creeper / skeleton / zombie code and removed the `BlockAbove` checks from it.

**Comment by Seadragon91** [comment]
Corrected it. Yes the code from zombie is better.

**Comment by Seadragon91** [comment]
Rebased on master.

**Pull request merged**